storage/myisam/mi_check.c

```diff
@@ -3429,9 +3429,9 @@ static int sort_get_next_record(MI_SORT_PARAM *sort_param)
 	  {
 	    if (!searching)
 	      mi_check_print_info(param,
-				  "Found block with impossible length %u at %s; Skipped",
-				  block_info.block_len+ (uint) (block_info.filepos-pos),
-				  llstr(pos,llbuff));
+                "Found block with impossible length %lu at %s; Skipped",
+                block_info.block_len + (unsigned long) (block_info.filepos-pos),
```

Comment:
`MI_BLOCK_INFO::filepos` and local var `pos` are both `my_off_t` which can be as large as `unsigned long long`, yet the old code casted to  `uint`.

Here I cast to `unsigned long` to match `MI_BLOCK_INFO::block_len`.